### PR TITLE
Fixed TV Recording channel for 10.9 

### PR DESCRIPTION
**Pull request by lomion0815**
Problem: 
- When registering LiveTvService and RecordingsChannel as Service we get a reference loop

Solution:
- Registered additional Service for HTSConnectionHandler
- Removed initialization of HTSConnectionHandler from LiveTvService
- Changed RecordingsChannel to not use LiveTvService but HTSConnectionHandler directly

### CODE REVIEW

File: TVHeadEnd/RecordingsChannel.cs [hunk 1]
```diff
@@ -140,7 +149,36 @@ public bool IsEnabledFor(string userId)
...
+        public async Task<IEnumerable<MyRecordingInfo>> GetAllRecordingsAsync(CancellationToken cancellationToken)
+        {
+            // retrieve all 'Pending', 'Inprogress' and 'Completed' recordings
+            // we don't deliver the 'Pending' recordings
+
+            int timeOut = await WaitForInitialLoadTask(cancellationToken);
+            if (timeOut == -1 || cancellationToken.IsCancellationRequested)
+            {
+                _logger.LogDebug("[TVHclient] GetAllRecordingsAsync - Not initialized ");
+                return new List<MyRecordingInfo>();
```
Comment (crobibero) [inline TVHeadEnd/RecordingsChannel.cs:168]:
```suggestion
                return [];
```

File: TVHeadEnd/RecordingsChannel.cs [hunk 2]
```diff
@@ -140,7 +149,36 @@ public bool IsEnabledFor(string userId)
...
+            }
+
+            TaskWithTimeoutRunner<IEnumerable<MyRecordingInfo>> twtr = new TaskWithTimeoutRunner<IEnumerable<MyRecordingInfo>>(TIMEOUT);
+            TaskWithTimeoutResult<IEnumerable<MyRecordingInfo>> twtRes = await
+                twtr.RunWithTimeout(_htsConnectionHandler.BuildDvrInfos(cancellationToken));
+
+            if (twtRes.HasTimeout)
+            {
+                _logger.LogDebug("[TVHclient] GetAllRecordingsAsync - Timeout");
+                return new List<MyRecordingInfo>();
```
Comment (crobibero) [inline TVHeadEnd/RecordingsChannel.cs:178]:
```suggestion
                return [];
```

File: TVHeadEnd/RecordingsChannel.cs [hunk 3]
```diff
@@ -140,7 +149,36 @@ public bool IsEnabledFor(string userId)
 
         private LiveTvService GetService()
         {
-            return _liveTvManager.Services.OfType<LiveTvService>().First();
+            return _htsConnectionHandler.getLiveTvService();
+        }
+
+        private Task<int> WaitForInitialLoadTask(CancellationToken cancellationToken)
+        {
+            return Task.Factory.StartNew<int>(() => _htsConnectionHandler.WaitForInitialLoad(cancellationToken));
```
Comment (crobibero) [inline TVHeadEnd/RecordingsChannel.cs:157]:
```suggestion
            return Task.Factory.StartNew<int>(() => _htsConnectionHandler.WaitForInitialLoad(cancellationToken), cancellationToken);
```

**Comment by lomion0815** [comment]
Thanks for the review. I commited your comments.

**Review by crobibero — APPROVED** [review]

**Pull request merged**